napari/utils/translations.py

```diff
@@ -0,0 +1,429 @@
+    dict
+        A dict with the native and display language for all locales found.
+
+    Example
+    -------
+    >>> get_language_packs("es_CO")
+    {
+        'en': {'displayName': 'Inglés', 'nativeName': 'English'},
+        'es_CO': {
+            'displayName': 'Español (colombia)',
+            'nativeName': 'Español (colombia)',
+        },
+    }
+    """
+    from napari_plugin_engine.manager import iter_available_plugins
+
+    lang_packs = iter_available_plugins(NAPARI_LANGUAGEPACK_ENTRY)
+    found_locales = {k: v for (k, v, _) in lang_packs}
+
+    invalid_locales = []
+    valid_locales = []
+    messages = []
+    for locale in found_locales:
+        if _is_valid_locale(locale):
+            valid_locales.append(locale)
+        else:
+            invalid_locales.append(locale)
+
+    display_locale = (
+        display_locale if display_locale in valid_locales else DEFAULT_LOCALE
+    )
+    locales = {
+        DEFAULT_LOCALE: {
+            "displayName": _get_display_name(DEFAULT_LOCALE, display_locale),
+            "nativeName": _get_display_name(DEFAULT_LOCALE, DEFAULT_LOCALE),
+        }
+    }
+    for locale in valid_locales:
+        locales[locale] = {
+            "displayName": _get_display_name(locale, display_locale),
+            "nativeName": _get_display_name(locale, locale),
+        }
+
+    if invalid_locales:
+        messages.append(
+            f"The following locales are invalid: {invalid_locales}!"
+        )
+
+    return locales
+
+
+# --- Translators
+# ----------------------------------------------------------------------------
+class TranslationBundle:
+    """
+    Translation bundle providing gettext translation functionality.
+
+
+    Parameters
+    ----------
+    domain : str
+        The python package/module that this bundle points to. This corresponds
+        to the module name of either the core package (``napari``) or any
+        extension, for example ``napari_console``. The language packs will
+        contain ``*.mo`` files with these names.
+    locale : str
+        The locale for this bundle. Examples include "en_US", "en_CO".
+    """
+
+    def __init__(self, domain: str, locale: str):
+        self._domain = domain
+        self._locale = locale
+
+        self._update_locale(locale)
+
+    def _update_locale(self, locale: str):
+        """
+        Update the locale environment variables.
+
+        Parameters
+        ----------
+        locale : str
+            The language name to use.
+        """
+        self._locale = locale
+        localedir = None
+        if locale.split("_")[0] != DEFAULT_LOCALE:
+            from napari_plugin_engine.manager import iter_available_plugins
+
+            lang_packs = iter_available_plugins(NAPARI_LANGUAGEPACK_ENTRY)
+            data = {k: v for (k, v, _) in lang_packs}
+            if locale not in data:
+                import warnings
+
+                warnings.warn("Requested locale not available: {locale}")
```

Comment:
```suggestion
                warnings.warn(f"Requested locale not available: {locale}")
```

Reply:
not sure if this needs an `f`. I won't let it hold up merge though, as it is easy to fix on your next localization PR